scripts/viz/feature_activation_plots.R

```diff
@@ -0,0 +1,132 @@
+
+palette_order <- c("BRCA", "PRAD", "TGCT", "KICH", "KIRP", "KIRC", "BLCA",
+                   "OV", "UCS", "CESC", "UCEC", "THCA", "PCPG", "ACC", "SKCM",
+                   "UVM", "HNSC", "SARC", "ESCA", "STAD", "COAD", "READ",
+                   "CHOL", "PAAD", "LIHC", "DLBC", "MESO", "LUSC", "LUAD",
+                   "GBM", "LGG", "LAML", "THYM", NA)
+
+plot_vae <- function(x, y, covariate, legend_show = TRUE) {
+  # Helper function to plot given coordinates and color by specific covariates
+  x_coord <- paste(x)
+  y_coord <- paste(y)
+  color_ <- covariate
+
+  p <- ggplot(combined_df,
+              aes_string(x = combined_df[[x_coord]],
+                         y = combined_df[[y_coord]],
+                         color = color_)) + 
+    geom_point() +
+    xlab(paste("encoding", x_coord)) +
+    ylab(paste("encoding", y_coord)) +
+    theme_bw() +
+    theme(text = element_text(size = 20))
+  
+  if (color_ == "acronym") {
+    p <- p + scale_colour_manual(limits = tcga_colors$`Study Abbreviation`,
+                                 values = tcga_colors$`Hex Colors`,
+                                 na.value = "black",
+                                 breaks = palette_order)
+  }
+
+  if (!legend_show) {
+    p <- p + theme(legend.position = "none")
+  }
+  print(p)
+}
+
+gender_encodings <- file.path("figures", "gender_encodings.pdf")
+gender_encodings_legend <- file.path("figures", "gender_encodings_legend.pdf")
+sample_type_file <- file.path("figures", "sample_type_encodings.pdf")
+sample_legend_file <- file.path("figures", "sample_type_encodings_legend.pdf")
+
+# Output plots with and without legend
+plot_vae(x = 82, y = 85, covariate = "gender", legend_show = FALSE)
+ggsave(gender_encodings, width = 6, height = 5)
+
+plot_vae(x = 82, y = 85, covariate = "gender", legend_show = TRUE)
+ggsave(gender_encodings_legend, width = 6, height = 5)
+
+# Subset new dataframe for visualizing Melanoma activation
+met_df <- combined_df %>%
+  dplyr::select("53", "66", "sample_type", "acronym") %>%
+  dplyr::mutate(label = paste0(sample_type, acronym))
+
+# Change labels to capture melanoma vs. non-melanoma
+met_df$label[(met_df$acronym == "SKCM") &
+               (met_df$sample_type != "Metastatic")] <- "Non-metastatic SKCM"
+met_df$label[(met_df$acronym != "SKCM") &
+               (met_df$sample_type != "Metastatic")] <- "Non-metastatic Other"
+met_df$label[(met_df$acronym != "SKCM") &
+               (met_df$sample_type == "Metastatic")] <- "Metastatic Other"
+met_df$label[(met_df$label == "MetastaticSKCM")] <- "Metastatic SKCM"
+
+p <- ggplot(met_df, aes(x = met_df$`53`, y = met_df$`66`, color = label)) + 
+  geom_point() +
+  xlab(paste("encoding 53")) +
+  ylab(paste("encoding 66")) +
+  theme_bw() +
+  theme(text = element_text(size = 20))
+
+# Save plots with and without legend
+p + theme(legend.position = "none")
+ggsave(sample_type_file, width = 6, height = 5)
+
+p
+ggsave(sample_legend_file, width = 6, height = 5)
```

Comment:
I think `ggsave(sample_legend_file, plot = p, width = 6, height = 5)` would be cleaner here

Reply:
:+1: